core/src/bytecode/ast/mod.rs

```diff
@@ -33,13 +33,12 @@ use typ::*;
 
 /// A Nickel AST. Contains a root node and a span.
 ///
-/// Both members are references so that `Ast` is cheap to clone - it's in fact `Copy` - and to
-/// store. Additionally, we don't expect to access the span much on the happy path, so the
-/// indirection should be ok.
-#[derive(Clone, Copy, Debug, PartialEq)]
+//TODO: we don't expect to access the span much on the happy path. Should we add an indirection
+//through a reference?
```

Comment:
Given that we mostly work with `&Ast` instead of `Ast`, I suspect that having `TermPos` inline doesn't hurt.

Reply:
Yeah, probably. Though it does take up space, for example in `[Ast]`. I mean, the positions have to leave somewhere anyway, but  more generally question is: if we don't need them in CPU cache, should we put them somewhere else? It could be in a dedicated, separate arena for example, so that reading and traversing ASTs don't need to bring them into cache.